src/whylogs/io/local_dataset.py

```diff
@@ -0,0 +1,104 @@
+import os
+from typing import Any, Callable, List, Optional, Tuple
+import abc
+from PIL import Image
+
+EXTENSIONS = ('.csv', '.xls', '.jpg', '.jpeg', '.png', '.ppm', '.bmp',
+              '.pgm', '.tif', '.tiff', '.webp')
+
+
+def valid_file(fname):
+    extension = os.path.splitext(fname)[1]
+    return extension in EXTENSIONS
+
+
+def file_loader(path: str) -> Any:
+
+    check_extensions = os.path.splitext(path)[1]
+
+    try:
+        with open(path, 'rb') as f:
+            img = Image.open(f)
+            return img, img.format
+    except Exception as e:
+        raise(e)
+
+
+class Dataset(abc.ABC):
+
+    def __init__(self,
+                 root_folder: str = "",
+                 feature_transforms: Optional[List[Callable]] = None,
+                 )->None:
+        self.root_folder = os.path.expanduser(root_folder)
+        self.feature_transforms = feature_transforms
+
+    def __getitem__(self, index: int) -> Any:
+        raise NotImplementedError
+
+    def __len__(self)-> int:
+        raise NotImplementedError
+
+    def __repr__(self)-> str:
+
+        head = "Dataset " + self.__class__.__name__
+        body = ["Number of files: {}".format(self.__len__())]
+        if self.root is not None:
+            body.append("Folder location: {}".format(self.root))
+        body += self.extra_repr().splitlines()
+        if hasattr(self, "transforms") and self.transforms is not None:
+            body += [repr(self.transforms)]
+        lines = [head] + [" " * self._repr_indent + line for line in body]
+        return '\n'.join(lines)
+
+
+class LocalDataset(Dataset):
+
+    def __init__(
+        self,
+        root_folder,
+        loader: Callable[[str], Any] = file_loader,
```

Comment:
this implies heavy coupling with image logic?

Reply:
any file, will start with images for now.